agent/api/container/status/managedagentstatus.go

```diff
@@ -0,0 +1,109 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package status
+
+import (
+	"errors"
+	"strings"
+)
+
+const (
```

Comment:
while I like using the iota/enum in Go for certain situations, Im not sure it makes sense here since we need to frequently marshal these values in and out of json, which means we have to define custom Marshal and Unmarshal functions.

My prefrence would be to just define these as string constants to avoid all the mapping and custom json marshalling, like:

```go
type ManagedAgentStatus string

const (
	// ManagedAgentStatusNone is the zero state of a managed agent
	ManagedAgentStatusNone = "NONE"
	// ManagedAgentCreated represents a managed agent which has dependencies in place
	ManagedAgentCreated = "CREATED"
	// ManagedAgentRunning represents a managed agent that has started sucessfully
	ManagedAgentRunning = "RUNNING"
	// ManagedAgentStopped represents a managed agent that has stopped
	ManagedAgentStopped = "STOPPED"
)
```

Reply:
For this as well as the Marshal/Unmarshal, I'm following the pattern in our other state files for consistency.

Reply:
There's only one other file doing this isn't there? doesn't seem like enough of a precedent to follow a poor code example

Reply:
https://github.com/aws/amazon-ecs-agent/search?q=iota <- check out how often status and `iota` happen.  This is a pattern.  And many of these end up with our custom marshal/unmarshal.

Reply:
Fair enough, I only looked in the same directory so didn't see all of these.

I think most of this code is just following a pattern that was defined many years ago -- I would still prefer to define a new pattern that can be used for future statuses and that hopefully could be used as an example to migrate the old ones to. 

IMHO the less custom json marshalling the better :)

Reply:
I agree that this custom marshaling is unnecessary complexity. I guess we can refactor that either now or later, not a blocker for this PR though